lib/android/app/src/main/java/com/reactnativenavigation/viewcontrollers/ModalStack.java

```diff
@@ -26,25 +27,33 @@
     void showModal(final ViewController viewController, Promise promise) {
         Modal modal = creator.create(viewController, this);
         modals.add(modal);
-		modal.show();
+        modal.show();
         promise.resolve(viewController.getId());
-	}
-
-	void dismissModal(final String componentId, Promise promise) {
-		Modal modal = findModalByComponentId(componentId);
-		if (modal != null) {
-			modal.dismiss(promise);
-		} else {
-			Navigator.rejectPromise(promise);
-		}
-	}
-
-	void dismissAll(Promise promise) {
-		for (Modal modal : modals) {
-			modal.dismiss(size() == 1 ? promise : new NoOpPromise());
-		}
-		modals.clear();
-	}
+    }
+
+    void dismissModal(final String componentId, Promise promise) {
+        Modal modal = findModalByComponentId(componentId);
```

Comment:
Would be nice to add a `applyOnModal` method which will accept three parameters,
1. componentId
2. accept method
3. reject method

This way we avoid the messy null checks